py-polars/tests/unit/namespaces/test_strptime.py

```diff
@@ -473,6 +473,24 @@ def test_to_datetime_ambiguous_or_non_existent() -> None:
         pl.Series(["2021-03-28 02:30"]).str.to_datetime(
             time_unit="us", time_zone="Europe/Warsaw"
         )
+    with pytest.raises(
+        pl.ComputeError,
+        match="datetime '2021-03-28 02:30:00' is non-existent in time zone 'Europe/Warsaw'",
+    ):
+        pl.Series(["2021-03-28 02:30"]).str.to_datetime(
+            time_unit="us",
+            time_zone="Europe/Warsaw",
+            ambiguous="null",
+        )
```

Comment:
separately, a `non_existent` argument could be added to make sure that this would return `null` values instead of raising. This is just to check that `ambiguous='null'` doesn't affect what happens for non-existent datetimes (coverage purposes)